core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/NormalizeOrInExpressionRewriter.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql.planner.iterative.rule;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMultimap;
+import io.trino.sql.tree.ComparisonExpression;
+import io.trino.sql.tree.Expression;
+import io.trino.sql.tree.ExpressionRewriter;
+import io.trino.sql.tree.ExpressionTreeRewriter;
+import io.trino.sql.tree.InListExpression;
+import io.trino.sql.tree.InPredicate;
+import io.trino.sql.tree.LogicalExpression;
+
+import java.util.LinkedHashSet;
+import java.util.List;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.trino.sql.ExpressionUtils.and;
+import static io.trino.sql.ExpressionUtils.or;
+import static io.trino.sql.tree.ComparisonExpression.Operator.EQUAL;
+import static io.trino.sql.tree.LogicalExpression.Operator.AND;
+
+public final class NormalizeOrInExpressionRewriter
```

Comment:
I add a new `NormalizeOrInExpressionRewriter` instead rename from `NormalizeOrExpressionRewriter`. It will merge: inList expression and comparison(EQUAL) expression, if possible.
If we remove `NormalizeOrExpressionRewriter` the only difference is the case `I1 = 1 OR I1 = 2 OR I1 IN (3, 4)` will be rewritten as `I1 IN (1, 2, 3, 4)` while current is `I1 IN (3, 4, 1, 2)`